D2L.Services.Core.WebApi/Service.cs

```diff
@@ -42,6 +43,30 @@ Type dependencyLoaderType
 				)
 			);
 		}
+		
+		public Service(

```

Comment:
It still needs the type in order to construct the inner dependency loader, which is why I have both a TestDependencyRegistry and an internal TestDependency Resolver.

If I made it take in an IDependencyResolver instead of a type, then the caller would have to construct the real resolver itself and manually add the dependencies that CoreDependencyLoader would have added (the config viewer and log provider), then wrap that in a TestDependencyResolver. That's not _really_ a big problem, but it means that you'd have to make another test dependency loader to add those 2 dependencies, then wrap that in the test dependency loader.

By just making the TestDependencyRegistry another layer ontop of all that, you don't have to worry about the inner workings of the dependency loaders and can just specify overrides instead.